waku/v2/protocol/waku_store/waku_store_types.nim

```diff
@@ -52,9 +52,15 @@ type
     startTime*: float64 # used for time-window query
     endTime*: float64 # used for time-window query
 
+  HistoryResponseError* {.pure.} = enum
+    ## HistoryResponseError contains error message to inform  the querying node about the state of its request
+    NONE = uint32(0)
+    INVALID_CURSOR = uint32(1)
+
   HistoryResponse* = object
     messages*: seq[WakuMessage]
     pagingInfo*: PagingInfo # used for pagination
+    error*: HistoryResponseError
```

Comment:
Maybe I'm still missing something, but if `HistoryResponseError` is `NONE`, couldn't that intent be communicated with `error*: Option[HistoryResponsError]`?

Reply:
In protobuf terms, that field would be emitted, so the equivalent thing (`0`) would go over the wire, I believe.

Reply:
>Maybe I'm still missing something, but if HistoryResponseError is NONE, couldn't that intent be communicated with error*: Option[HistoryResponsError]? Right now, the way I'm reading it, it seems like an ad-hoc implementation of Option type.

If we want the error codes to include the successful responses, then the `error` field should not be optional. Otherwise, this can be converted to an optional field. 

I suggest not to make the `error` field optional (although in protobuf fields are optional by default).  The reason is to make sure that the `error` is not omitted by mistake rather the response has indeed been concluded successfully.

Let me know whether this makes sense or shall I convert it to optional (or make it a simple string error). I'll follow your discretion.